### PR TITLE
Fix for: Forgotten Password Form Leaks Valid Usernames [TRUNK-3933]

### PR DESCRIPTION
**Pull request by bholagabbar**
Fixed the leakage of usernames in the legacy UI. If the user enters an invalid username, he will be assigned a random fake question on the basis of the hash-value of his username whose answer will always be false. The issue of no secret question/answer prompts an error.

### CODE REVIEW

**Comment by bholagabbar** [comment]
Sure. But I don't know how to..

**Comment by dkayiwa** [comment]
@bholagabbar did you get a chance to look at the existing controller unit tests in the module?

**Comment by bholagabbar** [comment]
Can you guide me a little on that?

**Comment by bholagabbar** [comment]
Yes sorry. I basically have to create a test module for this one as well. Ok will will try tomorrow and get back to you. Thanks

**Comment by bholagabbar** [comment]
Why are there test file with same name but different content in legacy ui and legacy ui-mod? Also, do I have to generate the JUnit XML file? Sorry if my question is stupid..never done this before :)

**Comment by dkayiwa** [comment]
Are you looking at controller tests like these? https://github.com/openmrs/openmrs-module-legacyui/tree/master/omod/src/test/java/org/openmrs/web/controller

Did you glance through these links?
https://wiki.openmrs.org/display/docs/Unit+Tests
https://wiki.openmrs.org/display/docs/Module+Unit+Testing
https://wiki.openmrs.org/display/docs/Unit+Testing+Conventions
https://wiki.openmrs.org/display/docs/Unit+Testing+with+Database+Data
https://wiki.openmrs.org/display/docs/Unit+Testing+With+at-should+Annotation
https://wiki.openmrs.org/display/docs/Generate+Test+Case+Plugin

File: omod/src/main/java/org/openmrs/web/controller/ForgotPasswordFormController.java [hunk 1]
```diff
@@ -124,11 +128,14 @@ protected ModelAndView onSubmit(HttpServletRequest request, HttpServletResponse
 					Context.removeProxyPrivilege(PrivilegeConstants.GET_USERS);
 				}
 				
-				if (user == null /*|| StringUtils.isEmpty(user.getSecretQuestion())*/) {
+				if (user == null) {
+					httpSession.setAttribute(WebConstants.OPENMRS_ERROR_ATTR, "auth.question.fill");
+					request.setAttribute("secretQuestion", getRandomFakeSecretQuestion(username));
```
Comment (wluyima) [inline omod/src/main/java/org/openmrs/web/controller/ForgotPasswordFormController.java]: Why ask them a fake question? I would just keep prompting them for the username? Is this the general behavior on other sites?

Comment (bholagabbar) [inline omod/src/main/java/org/openmrs/web/controller/ForgotPasswordFormController.java]: It was asked to implement it this way as it was mentioned in the ticket
comments. Is that alright?

Comment (wluyima) [inline omod/src/main/java/org/openmrs/web/controller/ForgotPasswordFormController.java]: Ok

**Pull request closed without merging**